apiserver/apiserver.go

```diff
@@ -1020,7 +998,7 @@ func (srv *Server) endpoints() ([]apihttp.Endpoint, error) {
 		pattern:    charmsObjectsRoutePrefix,
 		methods:    []string{"PUT"},
 		handler:    modelObjectsCharmsHTTPHandler,
-		authorizer: modelCharmsUploadAuthorizer,
+		authorizer: charmsObjectsAuthorizer,
```

Comment:
Why is this?

Reply:
Because it's only for objects, charms in this case. We should really generalise these.